src/ocamlorg_frontend/pages/package_versions.eml

```diff
@@ -0,0 +1,64 @@
+let render (package : Package.package)
+=
+Layout.render
+~title:(Printf.sprintf "%s Versions" package.name)
+~description:"Package Versions" @@
+<div class="bg-default dark:bg-dark-default py-8">
+  <div class="container-fluid flex items-center gap-2">
+    <a href="/p/<%s package.name %>" class="h-11 w-11 m-5 border-primary-600 border-2 rounded-full flex items-center justify-center"><%s! Icons.arrow_left "h-6 w-6 text-primary-600"; %></a>
+    <h1 class="text-2xl font-medium"><%s package.name %> Versions </h1><div class="text-2xl">(<%i List.length package.versions %>)</div>
+  </div>
+  <div class="container-fluid flex justify-center">
+    <table class="table-auto border-separate border-spacing-y-6">
+        <thead class="container-fluid text-md text-lighter text-left">
+            <tr class="py-4">
+                <th class="hidden sm:table-cell text-left font-normal border-b-2 border-gray-200 py-4 pl-3 md:pr-14"><%s! Icons.bars_3 "h-5 w-5"; %></th>
+                <th class="text-left font-normal border-b-2 border-gray-200 px-5 md:pr-12">Version</th>
+                <th class="text-left font-normal border-b-2 border-gray-200 md:pr-14">Release Date</th>
+                <th class="text-left font-normal border-b-2 border-gray-200 px-5 md:pr-14">Links</th>
+            </tr>
+        </thead>
+        <tbody>
+          <% package.versions |> List.iter (fun (version : string) -> %>
+          <tr class="tr-selected">
+            <% if List.nth package.versions 0 = version then ( %>
+            <td class="hidden sm:table-cell text-left font-normal py-5 px-4 bg-primary-200 rounded-l">
+              <div class="bg-primary-600 h-3 w-3 rounded-full"></div>
+              <div class="absolute h-[4.5rem] pl-1">
+                <div class="-ml-px w-1 border-solid border-r-2 border-gray-200 h-full"></div>
+              </div>
+            </td>
+            <td class="text-left bg-primary-200">
+              <a href="/p/<%s package.name %>/<%s version %>" class="p-5 hover:no-underline hover:text-primary-600 focus:text-primary-600 focus:no-underline font-bold focus:font-bold text-base md:mr-10"><%s version %></a>
+            </td>
+            <td class="pr-4 text-left text-sm bg-primary-200">
+              <%s Utils.human_date_of_timestamp package.publication %>
```

Comment:
Ah sorry, I overlooked that. To render the publication dates of the other versions, we need to look them up in `handler.ml` and pass them to the template as an argument.

This looks like it will require some changes in `ocamlorg_package.ml` and `ocamlorg_package.mli`. For example, we could add a function `get_versions_with_publication_date : state -> string -> (Version.t, float) list`, which, given the package state and the name of a package returns a list of all the versions and respective publication dates of the package.

Reply:
``get_versions`` in ``ocamlorg_package.ml`` seems pretty close to what i need but i'm not sure how to get the publication dates

Reply:
yes, for the new `get_versions_with_publication_date` function, I would start with what `get_versions` does and then try to change that to also give the publication date `float` value.

Reply:
Right, that's what my question was. Do you have any more specific guidance?

Reply:
You should also look at the function above `get_versions`. It is called `get_by_name` and retrieves the name, version, and package info for a package. The package info contains the publication date.

As a hint: the function `Version.Map.bindings` returns a list of key-value pairs `(version, info)`.